tests/test_memory_size.cpp

```diff
@@ -109,8 +109,9 @@ TEST_P(TestBadMemorySizeFormats, rejectsBadFormats)
 
 INSTANTIATE_TEST_SUITE_P(MemorySize, TestGoodMemorySizeFormats, ValuesIn(TestGoodMemorySizeFormats::generate_args()));
 INSTANTIATE_TEST_SUITE_P(MemorySize, TestBadMemorySizeFormats,
-                         Values("321BB", "321BK", "1024MM", "1024KM", "1024GK", "K", "", "123.321K", "123.321", "6868i",
-                                "555iB", "486ki", "54Mi", "8i33", "4M2", "-2345", "-5MiB", "K", "4GM"));
+                         Values("321BB", "321BK", "1024MM", "1024KM", "1024GK", "K", "", "123.321", "6868i", "555iB",
+                                "486ki", "54Mi", "8i33", "4M2", "-2345", "-5MiB", "K", "4GM", "256.M", "186000.B",
+                                "3.14", ".5g", "4.2B", "42.", "2048.K", " 268. "));
```

Comment:
While you're at it, why not accept `1.` and `.1` too? They're usually acceptable ways to input floats. You would have to protect the regexp against just `.` though. This is less relevant though, so leave it alone if you find it too troublesome.